minorminer/utils/feasibility.py

```diff
@@ -0,0 +1,234 @@
+            zephyr, pegasus or chimera and constructed by dwave_networkx.
+    Returns:
+        int: The maximum possible size of a tile
+    """
+    return max(T.graph.get("rows"), T.graph.get("columns"))
+
+
+def lattice_size_lower_bound(
+    S: nx.Graph,
+    T: nx.Graph = None,
+    topology: str = None,
+    t: int = None,
+    one_to_one: bool = False,
+) -> float:
+    """Returns a lower bound on the size necessary for embedding.
+
+    The lattice size is the parameter ``m`` of a dwave_networkx graph, also
+    called number of rows. The function returns a lower bound (necessary but
+    not sufficient for embedding) using efficiently established graph
+    properties such as the number of nodes, number of edges, node-degree
+    distribution, and two-colorability
+
+    Args:
+        S: The source graph to embed.
+        T: The target graph in which to embed. The graph must be of type
+            'zephyr', 'pegasus', or 'chimera' and constructed by dwave_networkx.
+        topology: The topology ('chimera', 'pegasus', or 'zephyr'). This is
+            inferred from `T` by default. Any set value must be consistent with
+            ``T`` (if ``T`` is not None).
+        t: The tile parameter, relevant for 'zephyr' and 'chimera' topologies.
+            Inferred from ``T`` by default. Any set value must be consistent with
+            ``T`` (if ``T`` is not None).
+        one_to_one: True if a subgraph embedding is assumed, False for general
+            embeddings.
+
+    Raises:
+        ValueError: If ``T`` and ``topology`` are inconsistent, or if ``T`` and ``t`` are inconsistent.
+        ValueError: If neither ``T`` nor ``topology`` is specified.
+        ValueError: If ``S`` cannot be embedded in ``T`` or the specified topology.
+
+    Returns:
+        float: Minimum ``lattice_size`` for embedding to be feasible. Returns
+        None if embedding for any number of graph rows is infeasible.
+    """
+    # Comment, similar bounds are possible allowing for minor embeddings,
+    # the degree distribution is bounded as chain length increases, whilst
+    # the number of nodes is depleted.
+    # This could be a possible feature expansion.
+    if T is not None:
+        if embedding_feasibility_filter(S, T, one_to_one) is False:
+            return None
+        
+        if topology is None:
+            topology = T.graph.get("family")
+        elif topology != T.graph.get("family"):
+            raise ValueError("Arguments T and topology are inconsistent")
+        
+        if t is None:
+            t = T.graph["tile"]
+    else:
+        if topology is None:
+            raise ValueError("T or topology must be specified")
+        
+        if t is None:
+            t = 4
+        max_degrees = {"chimera": 2 + 2 * t, "pegasus": 15, "zephyr": 4 + 4 * t}
+        max_source_degree = max(S.degree[n] for n in S.nodes())
+
+        if max_source_degree > max_degrees[topology]:
```

Comment:
This assumes `one_to_one` is True -- otherwise, this incorrectly reports that `nx.complete_graph(12)` is unembeddable into the 'chimera' topology.  The lower bound should be 3 in that case.  (confirmed with `embedding_feasibility_filter(nx.complete_graph(12), dnx.chimera_graph(3))`

Reply:
I've added the example above in a test case `test_lattice_size_lower_bound_one_to_one_false`